express/src/express.ts

```diff
@@ -177,5 +177,5 @@ function getController(Controller: Type): ExpressClass {
 function getParam(source: any, paramType: string, name: string): any {
   let param = source[paramType] || source;
 
-  return name ? param[name] : param;
+  return name ? param[name] : undefined;
```

Comment:
That's not really correct assumption, if `name` doesn't exist, `@Body`, `@Query` etc. should return entire object

Reply:
This is what happens in my second and third example, but I'm not sure I understand why the expectation is that this happens?
If I'm trying to fetch a certain query parameter but the request doesn't contain it, why would the decorator send me the whole object?

Reply:
your examples are correct, if you'll do `@Query('days')` it will return you undefined if it's no there, otherwise query object

Reply:
Never mind, I understand what you meant now. Using `@Query()` should return the whole object, not a particular property. Hm..

Reply:
exactly

Reply:
What I mean is:
If I use `@Query()`, I expect to get the entire query object.
If I use `@Query('days')`, I expect to get the value of the days property, if it was provided. If the request doesn't contain this query param, I expect the decorator to tell me that, not to return _the entire query object_.

Reply:
it shouldn't return you entire object, if name is available, that's what ternary operator does:
```
name ? param[name] : undefined
```